src/pages/ProductPage/ProductPage.js

```diff
@@ -1,29 +1,66 @@
-import React from "react";
-import { products } from "../../backend/db/products";
+import React, { useReducer } from "react";
 import "./ProductPage.css";
+import { defaultFilterState } from "../../reducer/defaultFilterState";
+import { filterReducer } from "../../reducer/filterReducer";
+import { getMinMaxPrice } from "../../utils/minMaxPrice";
+import { filterbySort } from "../../utils/filterbySort";
+import { filterByPriceRange } from "../../utils/filterByPriceRange";
+import { filterByRating } from "../../utils/filterByRating";
+import { filterByCategory } from "../../utils/filterByCategory";
+import { useEffect, useState } from "react";
 
 const ProductPage = () => {
+  const [state, dispatch] = useReducer(filterReducer, defaultFilterState);
+  const [products, setProducts] = useState([]);
+  useEffect(() => {
+    (() => {
+      fetch("/api/products")
+        .then((res) => res.json())
+        .then((data) => setProducts(data.products));
```

Comment:
Try to use `axios` instead of `fetch`. It has more features and is also simple to use.